rosdep/python.yaml

```diff
@@ -6319,6 +6328,7 @@ python3-grpcio:
   ubuntu:
     '*': [python3-grpcio]
     bionic: null
+python3-gsmmodem-pip: *migrate_eol_2025_04_30_python3_gsmmodem_pip
```

Comment:
Personally, I don't really think we need the migration key.  Python 2 is dead, so adding in new keys for it doesn't seem to make much sense.  So I would suggest just making the key `python3-gsmmodem-pip`, and dispensing with the `python-gsmmodem-pip` key and the migration anchor altogether.